### PR TITLE
Added VSCODE workspace and added formatting in CI

### PR DESCRIPTION
**Pull request by DipayanP007**
Solves #11 
* Created a VScode workspace file 
* set "format on save" as default setting
* Modified CI to run `cargo fmt --all`

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -65,9 +65,13 @@ jobs:
       run: |
         cd cli
         ./tests.sh
+    
+    - name: Check formatting
+      run: |
+           cargo fmt --all
```
Comment (torfmaster) [inline .github/workflows/build.yml]: I think you want to fail the CI, i.e. do a `cargo fmt --all -- --check`.

**Review by torfmaster — CHANGES_REQUESTED** [review]
Thank you for your PR! I have commented one minor topic but I am otherwise happy!

**Comment by DipayanP007** [comment]
Hey @torfmaster I made the changes kindly review it

**Comment by torfmaster** [comment]
@DipayanP007: The checks have failed (which shows that your change work :tada: )can you please format the code and check in the changes?

**Comment by torfmaster** [comment]
ping @DipayanP007

**Comment by DipayanP007** [comment]
Hey @torfmaster sorry I was a bit busy. Actually I don't have knowledge regarding Rust programming.

**Comment by torfmaster** [comment]
No worries. You just need to run `cargo fmt --all` and commit+push the result, this will fix the formatting errors.

**Comment by DipayanP007** [comment]
Hey @torfmaster I made the changes and pushed it. Give it a look

**Comment by torfmaster** [comment]
Thank you for your contribution!

**Pull request merged**